fs/filter/filter.go

```diff
@@ -433,7 +433,7 @@ func (f *Filter) MakeListR(ctx context.Context, NewObject func(ctx context.Conte
 		var (
 			checkers = ci.Checkers
 			remotes  = make(chan string, checkers)
-			g        errgroup.Group
+			g, gCtx  = errgroup.WithContext(ctx)
```

Comment:
If we are making a gCtx then we should use it instead of ctx in call to `NewObject` below

Reply:
This has been updated